### PR TITLE
Streaming newsfeed

### LINKED ISSUE #818: streaming newsfeed

### PR DESCRIPTION
**Pull request by Phanoix**
Adds a streaming newsfeed widget, similar to the streaming wire one.
resolves #818

### CODE REVIEW

**Comment by Phanoix** [comment]
@PhilAndrecheck need some help with language:
"There are new newsfeed items. Click here to load",
 "Stream Newsfeed items" (the title of the widget / main section of the page) -> or maybe this should just stay as it currently is (Your News feed)?

Since this is very prominent on the main / newsfeed page

**Comment by Phanoix** [comment]
Decided to use the language that's already in use for newsfeed and streaming wire.

File: mod/gc_streaming_content/views/default/css/css.php [hunk 1]
```diff
@@ -15,15 +15,15 @@
...
          cursor: pointer;
      }
      
-     .posts-holder ul:first-child{
+     .posts-holder, .newsfeed-posts-holder ul:first-child{
          border-left: 2px solid #9bc1d4;
      }
      
-     .posts-holder ul .list-break{
+     .posts-holder, .newsfeed-posts-holder ul .list-break{
```
Comment (piet0024) [inline mod/gc_streaming_content/views/default/css/css.php]: This is adding an extra border bottom on the wire.
Try:
.post-holder ul .list-break, .newsfeed-posts-holder ul .list-break

File: mod/gc_streaming_content/views/default/css/css.php [hunk 2]
```diff
@@ -15,15 +15,15 @@
...
          
      }  
-     .stream-new-wire:hover{
+     .stream-new-wire:hover, .stream-new-newsfeed:hover{
          text-decoration: underline;
          cursor: pointer;
      }
      
-     .posts-holder ul:first-child{
+     .posts-holder, .newsfeed-posts-holder ul:first-child{
```
Comment (piet0024) [inline mod/gc_streaming_content/views/default/css/css.php]: This adds a border to the river menu
Try two different styles
```
.posts-holder ul:first-child{
border-left: 2px solid #9bc1d4;
}

.newsfeed-posts-holder ul:first-child .panel-river{
border-left: 3px solid #9bc1d4;
}
```

File: mod/gc_streaming_content/views/default/js/stream_newsfeed.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                }
+            },
+            error: function(request, status, error) { 
+               console.log('error');
+
+            }
+        });
+}
+
+// is there a point to this?
```
Comment (piet0024) [inline mod/gc_streaming_content/views/default/js/stream_newsfeed.js]: Yes ... well no not really now that you mention it ...

**Pull request merged**